scripts/builtin/mark_down_creator.py

```diff
@@ -0,0 +1,385 @@
+
+
+        #Returns
+        mdFile.new_header(level=3, title="Returns")
+        if outputParameterCount > 0:
+            rows = int(len(outputParameters) / outputParameterCount)
+            mdFile.new_table(columns=outputParameterCount, rows=rows, text=outputParameters, text_align='center')
+        else:
+            for line in additionalInfos:
+                mdFile.new_paragraph(line)
+
+    mdFile.create_md_file()
+
+
+def cleanFileDataArray():
+    #Here I go through all elements of the file_data_array check if everything is correct and if not create a file  or meaningfull output
+    new_file_data_array = []
+    for entry in file_data_array:
+        missingValues = False
+        missingParameterString = ''
+
+        inputParameterCount = entry.param_count
+        I_HEADER_MISSING = entry.input_header_missing
+        I_DESCRIPTION_MISSING = entry.input_description_missing
+        outputParameterCount = entry.output_param_count
+        O_HEADER_MISSING = entry.output_header_missing
+        O_DESCRIPTION_MISSING = entry.output_description_missing
+        description = entry.description
+        fileName = entry.fileName
+
+
+        if len(description) == 0:
+            missingValues = True
+            missingParameterString += "There is no description of the function available!\n"
+
+        if I_HEADER_MISSING is True:
+            missingValues = True
+            missingParameterString += "There is either no input parameter header given or the syntax is incorrect!\n"
+
+        if I_DESCRIPTION_MISSING is True:
+            missingValues = True
+            missingParameterString += "There is either no dedicated Input section header or syntax is incorrect!\n"
+
+        if inputParameterCount == 0:
+            missingValues = True
+            missingParameterString += "There are either no parameters given within the file or syntax is incorrect!\n"
+
+        if O_DESCRIPTION_MISSING is True:
+            missingValues = True
+            missingParameterString += "There is either no dedicated Output section header or syntax is incorrect!\n"
+
+        if O_HEADER_MISSING is True:
+            missingValues = True
+            missingParameterString += "There is either no output parameter header given or the syntax is incorrect!\n"
+
+        if outputParameterCount == 0:
+            missingValues = True
+            missingParameterString += "There are either no output parameters given within the file or syntax is incorrect!\n"
+
+        if missingValues:
+            print('\033[93m' + "For the File: " + '\033[4m\033[92m' + fileName + '\033[0m' + '\033[93m' + " following errors occured:" + '\033[0m')
+            print(missingParameterString)
+        else:
+            new_file_data_array.append(entry)
+
+    for entry in incorrect_file_data:
+        print(f'{bcolors.WARNING}For the File: {bcolors.UNDERLINE}{bcolors.OKGREEN}' + entry + f'{bcolors.ENDC}{bcolors.WARNING} NO HEADER WAS FOUND AT ALL:{bcolors.ENDC}')
```

Comment:
Please use consistent error messages - not all caps

Reply:
Also it would be great to have an optional input argument to influence the error handling - if set to true, we could collect if there was at least one warning printed and if so throw an exception.